lib/t/matches.ex

```diff
@@ -307,6 +376,41 @@ defmodule T.Matches do
     end)
   end
 
+  defp with_expiration_date(matches) do
```

Comment:
This function lists all events for all matches, one by one, that's wasteful. Please check out lateral jons, it can be done in a single query when listing matches.